docs/content/config/advanced/auth-lua.md

```diff
@@ -0,0 +1,194 @@
+
+???+ example
+
+    DMS configuration environment variables to let Postfix apply Lua authentication by proxy through Dovecot for authenticated SMTP.
+
+    ```yaml
+    - ENABLE_SASLAUTHD=1
+    - SASLAUTHD_MECHANISMS=rimap
+    - SASLAUTHD_MECH_OPTIONS=127.0.0.1
+    ```
+
+## Modifying Dovecot's configuration
+
+Add the following volume values to the relevant part of `compose.yaml`:
+
+???+ example
+
+    Override and add Dovecot configuration files and lua scripts.
+
+    ```yaml
+        volumes:
+          - ./docker-data/dms/config/dovecot/auth-ldap.conf.ext:/etc/dovecot/conf.d/auth-ldap.conf.ext:ro
+          - ./docker-data/dms/config/dovecot/auth-lua-httpbasic.conf:/etc/dovecot/conf.d/auth-lua-httpbasic.conf:ro
+          - ./docker-data/dms/config/dovecot/lua/:/etc/dovecot/lua/:ro
+    ```
+
+The first volume line [overrides][docs-dovecotoverrideconfiguration] Dovecot's standard LDAP authentication configuration file. The second line [adds][docs-dovecotaddconfiguration] a new configuration file for Lua authentication. The third line adds a directory which will contain Lua scripts. The files and directory will not be changed from inside the container, which is why they are configured as read-only.
+
+Make the necessary changes on the filesystem (*where `mailserver` is the container name*):
+```bash
+mkdir -p ./docker-data/dms/config/dovecot/lua
+docker cp mailserver:/etc/dovecot/conf.d/auth-ldap.conf.ext ./docker-data/dms/config/dovecot/auth-ldap.conf.ext
+```
+
+Edit configuration file `./docker-data/dms/config/dovecot/auth-ldap.conf.ext`. Comment out the passdb section. An excerpt of what that part would look like after you are done:
+```
+#passdb {
+#  driver = ldap
+#
+#  # Path for LDAP configuration file, see example-config/dovecot-ldap.conf.ext
+#  args = /etc/dovecot/dovecot-ldap.conf.ext
+#}
+```
+
+Don't touch anything else in the file.
+
+Create configuration file `./docker-data/dms/config/dovecot/auth-lua-httpbasic.conf` with contents:
+```
+passdb {
+  driver = lua
+  args = file=/etc/dovecot/lua/auth-httpbasic.lua blocking=yes
+}
+```
+
+That is all for configuring Dovecot.
+
+## Create the Lua script
+
+Create Lua file `./docker-data/dms/config/dovecot/lua/auth-httpbasic.lua` with contents:
+
+```lua
+local http_url = "https://nextcloud.example.com/remote.php/dav/"
+local http_method = "PROPFIND"
+local http_status_ok = 207
+local http_status_failure = 401
+local http_header_forwarded_for = "X-Forwarded-For"
+
+package.path = package.path .. ";/etc/dovecot/lua/?.lua"
+local base64 = require("base64")
+
+local http_client = dovecot.http.client {
+    timeout = 1000;
+    max_attempts = 1;
+    debug = false;
+}
+
+function script_init()
+  return 0
+end
+
+function script_deinit()
+end
+
+function is_nextcloud_apppassword(password)
+  return string.find(password, "%w%w%w%w%w%-%w%w%w%w%w%-%w%w%w%w%w%-%w%w%w%w%w%-%w%w%w%w%w") ~= nil
+end
```

Comment:
**Optional**

Using `+` (_one or more_) and [makes the pattern a little easier to grok](https://gitspartv.github.io/lua-patterns/?pattern=%25w%2B-%25w%2B-%25w%2B-%25w%2B-%25w%2B). Although it's less specific (_[Lua pattern matching](https://riptutorial.com/lua/topic/5829/pattern-matching) doesn't seem to support repetition length like regex can_):

```suggestion
function is_nextcloud_apppassword(password)
  return string.find(password, "%w+-%w+-%w+-%w+-%w+") ~= nil
end
```